### DISCUSSION TITLE
How to decide the value of dt of generate_LOH_source_type50.py in WP2_LOH1?

### DISCUSSION THREAD

**Discussion by buzhiqiuyuebai** [SeisSol/SeisSol#867] [Q&A] [answered]
I find `dt =0.002` in [generate_LOH_source_type50.py](https://github.com/SeisSol/Examples/blob/59a3d4246e8cee4dd2b7b62e173a1507a31a1728/WP2_LOH1/generate_LOH_source_type50.py#L30). I Fourier transformed it and found that its cutoff frequency is 2500Hz, and even if I just take the value interval of (0,1) in the time domain, its spectrum is basically the same, because (1,4) interval values are close to 0, so the contribution to the spectrum is very small, according to Nyquist's sampling law, the sampling rate is at least twice the maximum cutoff frequency, so the sampling rate should be at least 5000, also That is, 5000 points should be taken between (0,1), so 5000*4=20,000 points are taken between (0,4). I have three questions:
1, is this sampling you are picking at twice the maximum cutoff frequency?
2, if you increase the number of points sampled, for example, 40,000 points will reduce the speed of calculation twice?
3, after the extra 0 on the speed of the calculation is not a big impact? For example, if I extend the signal application time to 8 s, until it is closer to 0, will this significantly slow down the calculation speed?

![image](https://github.com/SeisSol/SeisSol/assets/114474894/4b26137f-393d-47bc-91e7-d46c4c6f5764)
![image](https://github.com/SeisSol/SeisSol/assets/114474894/562dff9e-4308-4a9e-9f33-dd9e9cf5d329)
![image](https://github.com/SeisSol/SeisSol/assets/114474894/801cfdca-ccd7-4e2f-be16-ceb8661a5f7f)

**Comment by sebwolf-de** [comment]
The time step is calculated such that the CFL condition is met. We do not take the resolution of the source time function sampling into account while computing the time step.
Also, the quality of the solution (e.g. synthetic seismogram at a receiver) will depend much more on the resolution of the mesh in comparison to the resolution of the source time function. No feasible mesh (at the length scales of LOH1) will be able to resolve >1000Hz.
So: Resolution of the source time function does not change the computational efficiency and, if it is above a reasonable threshold, also doesn't inflict the quality of the result.

**Reply by buzhiqiuyuebai** [reply]
> The time step is calculated such that the CFL condition is met. We do not take the resolution of the source time function sampling into account while computing the time step. Also, the quality of the solution (e.g. synthetic seismogram at a receiver) will depend much more on the resolution of the mesh in comparison to the resolution of the source time function. No feasible mesh (at the length scales of LOH1) will be able to resolve >1000Hz. So: Resolution of the source time function does not change the computational efficiency and, if it is above a reasonable threshold, also doesn't inflict the quality of the result.

Does at least the number of sampling points at least satisfy Nyquist's sampling law? If the number of sampling points is too small does this result in discrete points that do not accurately represent the source function.

**Reply by sebwolf-de** [reply, marked as answer]
We do not have such checks and do not calculate any cutoff frequencies in SeisSol. You are right, that we then only simulate a low pass filtered version of the source time function, but:
* Typical time step sizes are in the order of 1e-4 to 1e-5, so we are pretty sure, we capture most of the spectrum of the source time function.
*  Typically, we don't resolve wavelengths smaller than 50 m (of course depending on the mesh). With typical S wave speeds, of 2000 m/s to 4000 m/s, the solution will contain frequencies up to 80 Hz. And this is already at the extreme! For realistic earthquake scenarios, 10Hz is a very high resolution. At this point, the sampling of the source time frequency has become irrelevant.

To conclude: If you need to resolve _all_ frequencies of the source time function, it is possible. But the mesh to resolve _all_ these frequencies will be huge (if not infeasible to create and run).
